packages/regl-worldview/src/WorldviewContext.js

```diff
@@ -240,63 +244,104 @@ export class WorldviewContext {
+              );
+              break;
+            } else if (counter > 0 && !enableStackedObjectEvents) {
+              // Only allow multiple object selection if we've explicitly enabled it.
+              break;
+            }
+            counter++;
+            regl.clear({ color: [0, 0, 0, 1], depth: 1 });
+            this._drawInput(true, seenObjects);
+
+            // it's possible to get x/y values outside the framebuffer size
+            // if the mouse quickly leaves the draw area during a read operation
+            // reading outside the bounds of the framebuffer causes errors
+            // and puts regl into a bad internal state.
+            // https://github.com/regl-project/regl/blob/28fbf71c871498c608d9ec741d47e34d44af0eb5/lib/read.js#L57
+            if (x < Math.floor(width) && y < Math.floor(height) && x >= 0 && y >= 0) {
+              const pixel = new Uint8Array(4);
+
+              // read pixel value from the frame buffer
+              regl.read({
+                x,
+                y,
+                width: 1,
+                height: 1,
+                data: pixel,
+              });
+
+              currentObjectId = getIdFromColor(pixel);
+              if (currentObjectId && this.getDrawPropByHitmapId(currentObjectId).object) {
+                seenIds.push(currentObjectId);
+                seenObjects.push(this.getDrawPropByHitmapId(currentObjectId));
+              }
+            }
+          } while (currentObjectId !== 0);
+          resolve(seenIds);
         });
       });
     });
   }
 
-  callComponentHandlers = (objectId: number, ray: Ray, e: MouseEvent, mouseEventName: MouseEventEnum) => {
-    this._hitmapCalls.forEach((_, component) => {
+  callComponentHandlers = (hitmapIds: Array<HitmapId>, ray: Ray, e: MouseEvent, mouseEventName: MouseEventEnum) => {
+    this._drawCalls.forEach((drawInput, component) => {
       if (component instanceof Command) {
-        component.handleMouseEvent(objectId, e, ray, mouseEventName);
+        const hitmapIdsForCommand = intersection(this._hitmapIdManager.getHitmapIdsForCommand(component), hitmapIds);
+        const mouseEvents = hitmapIdsForCommand.map((hitmapId) => this.getDrawPropByHitmapId(hitmapId));
+        component.handleMouseEvent(mouseEvents, e, ray, mouseEventName);
       }
     });
   };
 
-  _drawInput = (isHitmap?: boolean) => {
-    const drawCallsMap = isHitmap ? this._hitmapCalls : this._drawCalls;
-    const sortedDrawCalls = Array.from(drawCallsMap.values()).sort((a, b) => (a.layerIndex || 0) - (b.layerIndex || 0));
+  _drawInput = (isHitmap?: boolean, seenObjects?: Array<MouseEventObject>) => {
+    if (isHitmap) {
+      this._hitmapIdManager.reset();
+    }
 
+    const drawCalls = Array.from(this._drawCalls.values());
+    const sortedDrawCalls = drawCalls.sort((a, b) => (a.layerIndex || 0) - (b.layerIndex || 0));
```

Comment:
nit: while you are here, since sort is mutating, no need for 2 separate variables